src/mlpack/methods/ann/layer/transposed_convolution_impl.hpp

```diff
@@ -114,30 +114,39 @@ void TransposedConvolution<
     OutputDataType
 >::Forward(const arma::Mat<eT>&& input, arma::Mat<eT>&& output)
 {
-  inputTemp = arma::cube(input.memptr(), inputWidth, inputHeight, inSize);
+  batchSize = input.n_cols;
+  inputTemp = arma::cube(input.memptr(), inputWidth, inputHeight,
```

Comment:
hm, should we use ``copy_aux_mem = false`` here to avoid another copy?

Reply:
@zoq I don't think we can do that as that would cause it to cast away the constness of the input matrix.

Reply:
Good point, I guess we could do the same as done here:

https://github.com/mlpack/mlpack/blob/7fead71e00d8a6c3b9b2a4708a13cc469f259ffb/src/mlpack/methods/logistic_regression/logistic_regression_function_impl.hpp#L26-L27

your call.

Reply:
I'll make the changes, since it would lead to a considerable speedup for larger batch-sizes.